client/src/AppProvider.tsx

```diff
@@ -1,8 +1,11 @@
 import { CssBaseline } from '@mui/material';
 import { ThemeProvider, createTheme, Theme } from '@mui/material/styles';
+import RelayEnvironment from 'RelayEnvironment';
+import { AuthProvider } from 'components/AuthContext';
```

Comment:
There is already PrivateRoute component in index.tsx. Do we need this here? May be @KarstenMalle can also provide feedback here.

Reply:
This is not using any of our code so I can't really comment on it. 

We use our `AuthProvider` in `AppProvider.tsx` and then the route we want protected in index are wrapped with `PrivateRoute` as so:

```
 <PrivateRoute>
    <Library />
  </PrivateRoute>
```

Reply:
The dependency is from `PrivateRoute -> AuthContext -> Auth`. Is there a need to introduce `AuthContext` into `AppProvider`?

Reply:
I believe `PrivateRoute` can only be used within the context of `AuthProvider`. If removed, we get the following error:
```md
# Error
useAuth must be used within an AuthProvider
    at useAuth (http://localhost:3000/static/js/bundle.js:803:11)
    at PrivateRoute (http://localhost:3000/static/js/bundle.js:1244:60)
    at renderWithHooks (http://localhost:3000/static/js/bundle.js:64425:22)
    at mountIndeterminateComponent (http://localhost:3000/static/js/bundle.js:67711:17)
    at beginWork (http://localhost:3000/static/js/bundle.js:69007:20)
    at HTMLUnknownElement.callCallback (http://localhost:3000/static/js/bundle.js:54017:18)
    at Object.invokeGuardedCallbackDev (http://localhost:3000/static/js/bundle.js:54061:20)
    at invokeGuardedCallback (http://localhost:3000/static/js/bundle.js:54118:35)
    at beginWork$1 (http://localhost:3000/static/js/bundle.js:73992:11)
    at performUnitOfWork (http://localhost:3000/static/js/bundle.js:73239:16)
```

Correct me if I'm wrong @KarstenMalle, but the `authProvider` is the global context, which `PrivateRoute` uses? Regardless it follows the same structure as [893e130](https://github.com/prasadtalasila/DTaaS-Public/commit/893e130c0ef44b91b45cd35c335f6c23bb4bfc94) in PR #77, so I suggest moving this to a separate issue instead of this PR @prasadtalasila

Reply:
We might be misunderstanding each other. As can be seen in PR #77 `AuthContext` is not something we are using, and it is old code (over 2 months) as we are still waiting for the authentication PR to be approved. In the new code `AuthProvider` is the global context which informs `PrivateRoute` on if the user is authenticated or not.

Reply:
Ahhh thank you for the clarification! I realise we moved it in this PR to keep things streamlined. I reverted the change in commit [3093d76](https://github.com/INTO-CPS-Association/DTaaS/pull/78/commits/3093d762bc000f66f0cc491bc885badbeae19c66) and will resolve this comment.